Ra5 validation plots

folguera opened the pull request:
- Added some validation plots to be plotted for the baseline region. 

Major change:  
-  Now histograms have to be declared either global or local (true for global, false for local  - default is false): 
  _hm->addVariable("HT"    , 800, 0. , 800, "H_{T} [GeV]"                        , true);
- A histogram defined as global will be filled for every SR (workflow) 
- A histogram defined as local will be filled only once when the fill function is called (i.e. validation histograms are filled only once) 
- This was achieved by adding a data member to the hObs class and modifying the addVariable function in Histomanager

folguera commented:
Please, don't merge it yet, I'll add a card to produce the validation plots more or less automatically.

mmarionncern commented:
@folguera Santi, can you turn the public isgbl variable into a private one with access functions? (just for safety)

folguera commented:
@mmarionncern  Now everything works like a charm, could you please merge?  

I've added some missing plots, fixed normalisation problems by merging the latest PR from jmvizan and also adjusted the binning of the histograms. For now the whole package runs inside SSDL2015 but at some point we may want to create a standalone suite for doing this.

folguera commented:
@mmarionncern Now everything works like a charm, could you please merge?

I've added some missing plots, fixed normalisation problems by merging the latest PR from jmvizan and also adjusted the binning of the histograms. For now the whole package runs inside SSDL2015 but at some point we may want to create a standalone suite for doing this.

mmarionncern commented on display/core/DisplayClass.cc:
```diff
@@ -1176,8 +1176,8 @@ DisplayClass::prepareHistograms(const hObs* theobs) {
-  _empty->SetName("h_tmp");
-  _empty->SetTitle("h_tmp");
+  _empty->SetName("");
+  _empty->SetTitle("");
```
@folguera , what is the reason for removing the histogram title? if we have to change the plot by hand right before a paper submission, it will be more complicated to change the default histogram on the fly w/o name.

mmarionncern commented on display/core/MPAFDisplay.cc:
```diff
@@ -586,8 +586,8 @@ MPAFDisplay::getIntegral(float x1, float x2, float y1, float y2) {
-  // anConf.reset();
-  // _hm->reset();
+  anConf.reset();
+  _hm->reset();
```
Doing so allows to rerun the card interactively directly w/o exiting root?

folguera replied:
That was the intention but I've just tried and it doesn't seem to work... shall I change it back?

mmarionncern replied:
The reset function already takes care of cleanign everything, the refresh should not need to reinitialize the configuration

folguera commented:
@mmarionncern can you merge?
